src/hotspot/share/cds/heapShared.cpp

```diff
@@ -323,56 +323,42 @@ void HeapShared::restore_loader_data() {
   java_lang_ClassLoader::release_set_loader_data(SystemDictionary::java_system_loader(), _saved_java_system_loader_data);
 }
 
-class KlassToOopHandleTable: public ResourceHashtable<Klass*, OopHandle,
+class MetaspaceObjToOopHandleTable: public ResourceHashtable<MetaspaceObj*, OopHandle,
     36137, // prime number
     AnyObj::C_HEAP,
     mtClassShared> {
 public:
-  oop get_oop(Klass* k) {
+  oop get_oop(MetaspaceObj* obj) {
```

Comment:
The word `obj` may be confusing here as it may look like an oop. I think it's better to change it to `ptr`.

A bit of background:

CDS copies two types of objects:  oops (archiveHeapWriter.cpp and heapShared.cpp) and MetsapceObjs (in archiveBuilder.cpp) 

In the former two files, the convention is to use `obj` to refer to oops, and `ptr` to refer to MetsapceObjs.

In archiveBuilder.cpp, since we never use a mix of oops and MetsapceObjs, we use `obj` to refer to MetsapceObjs. This is mostly for historical purposes, and we might some day change it to use `ptr` to be more consistent.